### DISCUSSION TITLE
ReactiveComponent connect / Cannot read property 'endsWith' of undefined

### DISCUSSION THREAD

**Discussion by manuelabarca** [appbaseio/reactivesearch#1779] [Q&A] [answered]
I have a custom component with ReactiveComponent, it works 99%. I get the results without problems, but when I delete the first search and do a new search and press to trigger the search, I get the following error:

"Cannot read property 'endsWith' of undefined"

I can't figure out what error I may have ... or if it is a bug
![IMG_0168](https://user-images.githubusercontent.com/20046746/130989051-2fe5b279-1ddd-4782-b7b7-efa2827da0eb.PNG)

**Comment by bietkul** [comment, marked as answer]
@manuelabarca It seems like a usage issue. Can you share a reproducible example?

**Reply by manuelabarca** [reply]
It was my error, I just saw that I was missing the componentId to the reactivecomponent of the list, so it was always "undefined", taking advantage of this instance, I have a query like this

`{
            track_scores: true,
            sort: ["_score"],
            size: 20,
            from: 0,
            query: {
                bool: {
                    must: [
                        {
                            match: {
                                visible: 1,
                            },
                        },
                        {
                            match: {
                                name: {
                                    query: value,
                                    fuzziness: "AUTO",
                                    fuzzy_transpositions: "true",
                                    prefix_length: 1,
                                },
                            },
                        },
                    ],
                },
            },
            value,
        }`

But it always brings me 10 hits, is there a way to bring me paging? or remove that limitation, greetings and thank you very much in advance
